src/Backend/Modules/Settings/Js/Settings.js

```diff
@@ -140,7 +141,58 @@ jsBackend.settings =
         );
     },
 
-    startDotAnimation: function (speed, dotAmount)
+    generateSitemap: function (e)
+    {
+        // prevent default
+        e.preventDefault();
+
+        // save the button for later use
+        var $generateSitemapButton = $('[data-role="fork-generate-sitemap"]');
+
+        // disable the handler to prevent sending too many requests
+        $generateSitemapButton.off('click', jsBackend.settings.generateSitemap);
+        $generateSitemapButton.attr('disabled', 'disabled');
+
+        // display the status alert
+        var $statusAlert = $('[data-role="fork-generate-sitemap-status"]');
+        $statusAlert.toggleClass('hidden');
+
+        // start the dot animation
+        var dotAnimation = jsBackend.settings.startDotAnimation($statusAlert);
+
+        $.ajax(
+            {
+                timeout: 60000, // generating the sitemap might take a while
+                data: {
+                    fork: {
+                        module: 'SitemapGenerator',
+                        action: 'GenerateSitemap'
+                    }
+                },
+                success: function(data)
+                {
+                    jsBackend.messages.add('success', jsBackend.locale.msg('SitemapGenerated'));
+                },
+                error: function()
+                {
+                    // show error in case something goes wrong with the call itself
+                    jsBackend.messages.add('danger', jsBackend.locale.err('SomethingWentWrong'));
+                },
+                complete: function()
+                {
+                    // stop the dot animation
+                    jsBackend.settings.stopDotAnimation(dotAnimation, $statusAlert);
+                    // hide the status
+                    $statusAlert.toggleClass('hidden');
+                    // reset the button
+                    $generateSitemapButton.on('click', jsBackend.settings.generateSitemap);
+                    $generateSitemapButton.attr('disabled', false);
+                }
+            }
+        );
+    },
+
+    startDotAnimation: function ($element, speed, dotAmount)
```

Comment:
Isn't there a `loading` class for buttons?

Reply:
There is but this animation has nothing to do with the button